lib/puppet/util/windows/registry.rb

```diff
@@ -31,16 +31,24 @@ def open(name, path, mode = KEY_READ | KEY64, &block)
     end
 
     def values(subkey)
+      orig_enc = Encoding.default_internal
+      Encoding.default_internal = Encoding::UTF_8
```

Comment:
@Iristyle @kylog @joshcooper is doing this a good idea? I don't see another way around it. I could go with another encoding here if we think this will be the method but not sufficient. Looks like it removed the endash in favor of just a dash (but I'd need to inspect better to be sure).